File: String/28_ImplementstrStr.cpp
/*
 * @Author: xuezaigds@gmail.com
 * @Last Modified time: 2016-04-06 21:22:30
 */

class Solution {
public:
    int strStr(string haystack, string needle) {
        if(needle=="")  return 0;
        int str_len = haystack.size();
        int pat_len = needle.size();
        for(int i=0;i<str_len-pat_len+1;i++){
            int j=0;
            for(;j<pat_len;j++){
                if(haystack[i+j] != needle[j]){
                    break;
                }
            }
            if(j == pat_len){
                return i;
            }
        }
        return -1;
    }
};

/*
""
""
"abaa"
"aa"
"aaabbb"
"abbb"
*/

File: String/28_ImplementstrStr.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-


class Solution(object):
    # Not notoriously hard-to-understand algorithm KMP
    def strStr(self, haystack, needle):
        # Return 0 if needle is ""
        if not needle:
            return 0
        length = len(haystack)

        # If one char in haystack is same with needle[0],
        # then verify the other chars in needle.
        for i in range(length-len(needle)+1):
            if haystack[i:i+len(needle)] == needle:
                return i

        return -1

"""
""
""
"abaa"
"aa"
"aaabbb"
"abbb"
"""
